### DISCUSSION TITLE
How to debug using neotest (python adapter) with breakpoints.

### DISCUSSION THREAD

**Discussion by dbatten5** [nvim-neotest/neotest#339] [Q&A] [answered]
A common flow I used to use with `vim-test` was to insert a `breakpoint()` somewhere into the codebase, run the test, then it would drop me into a repl at the breakpoint so I can query things, check variable values, etc. If I insert a `breakpoint()` and run a test with `neotest` then test hangs and need to be manually stopped. I know I can integrate with `nvim-dap` to do this but just wondering if there's a way with `breakpoints()` too

**Comment by rcarriga** [comment, marked as answer]
You're looking for `:h neotest.run.attach()`, which opens up a terminal window for interacting with the running process. There is also a summary window mapping for it

**Reply by dbatten5** [reply]
Ah amazing, thanks @rcarriga !
